components/experimental/src/dimension/currency/long_format.rs

```diff
@@ -0,0 +1,127 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use core::str::FromStr;
+
+use fixed_decimal::FixedDecimal;
+
+use icu_decimal::FixedDecimalFormatter;
+use icu_pattern::DoublePlaceholderPattern;
+use icu_plurals::PluralRules;
+use writeable::Writeable;
+
+use crate::dimension::provider::currency_patterns::{CurrencyPatternsDataV1, PatternCount};
+use crate::dimension::provider::extended_currency::{Count, CurrencyExtendedDataV1};
+
+use super::CurrencyCode;
+
+pub struct LongFormattedCurrency<'l> {
+    pub(crate) value: &'l FixedDecimal,
+    pub(crate) currency_code: CurrencyCode,
+    pub(crate) extended: &'l CurrencyExtendedDataV1<'l>,
+    pub(crate) patterns: &'l CurrencyPatternsDataV1<'l>,
+    pub(crate) fixed_decimal_formatter: &'l FixedDecimalFormatter,
+    pub(crate) plural_rules: &'l PluralRules,
+}
+
+writeable::impl_display_with_writeable!(LongFormattedCurrency<'_>);
+
+impl<'l> Writeable for LongFormattedCurrency<'l> {
+    fn write_to<W>(&self, sink: &mut W) -> core::result::Result<(), core::fmt::Error>
+    where
+        W: core::fmt::Write + ?Sized,
+    {
+        let plural_category = self.plural_rules.category_for(self.value);
+        let count = Count::from(plural_category);
+        let pattern_count = PatternCount::from(plural_category);
+
+        let display_names = &self.extended.display_names;
+
+        let display_name = display_names
+            .get(&count)
+            .or_else(|| display_names.get(&Count::Other))
+            // TOOD: check the cases of using `Count::DisplayName`
+            .unwrap_or_else(|| self.currency_code.0.as_str());
+
+        let patterns = &self.patterns.unit_patterns;
+        let pattern = patterns
+            .get(&pattern_count)
+            .or_else(|| patterns.get(&PatternCount::Other))
+            .ok_or(core::fmt::Error)?;
```

Comment:
don't ever produce `core::fmt::Error`: https://doc.rust-lang.org/std/fmt/index.html#:~:text=Additionally%2C%20the%20return,up%20the%20stack.

Reply:
mmmmm

in this case we have two options,

1. to not return error at all and use default pattern
2. move this logic to `format_fixed_decimal` method 

I prefer option 2.

Reply:
Actually, `other` must be present. I added this condition to the datagen`

Reply:
https://github.com/unicode-org/icu4x/issues/5374

Reply:
I'd prefer if this was represented in the type system, instead of with a trust-me-unwrap.

Reply:
good point: https://github.com/unicode-org/icu4x/issues/5374#issuecomment-2291303219

Reply:
done :)